coral/src/app/layout/header/HeaderMenuLink.test.tsx

```diff
@@ -1,61 +1,101 @@
+import data from "@aivenio/aquarium/dist/src/icons/console";
+import { cleanup, render, screen } from "@testing-library/react";
+import userEvent from "@testing-library/user-event";
 import HeaderMenuLink from "src/app/layout/header/HeaderMenuLink";
-import { cleanup, screen, render, within } from "@testing-library/react";
-import data from "@aivenio/design-system/dist/src/icons/console";
+import { tabNavigateTo } from "src/services/test-utils/tabbing";
 
 // mock out svgs to avoid clutter
-jest.mock("@aivenio/design-system", () => {
+jest.mock("@aivenio/aquarium", () => {
   return {
     __esModule: true,
-    ...jest.requireActual("@aivenio/design-system"),
+    ...jest.requireActual("@aivenio/aquarium"),
 
     Icon: () => {
       return <div data-testid={"ds-icon"}></div>;
     },
   };
 });
 
+const linkText = "Go to your profile page";
+
 describe("HeaderMenuLink.tsx", () => {
   // (icon is not needed for the test, Icon component mocked out)
   const mockIcon = "" as unknown as typeof data;
 
   describe("renders a default link with required props", () => {
-    beforeAll(() => {
+    beforeEach(() => {
       render(
         <HeaderMenuLink
           icon={mockIcon}
           href={"/myProfile"}
-          linkText={"Go to your profile page"}
+          linkText={linkText}
         />
       );
     });
 
-    afterAll(cleanup);
+    afterEach(cleanup);
 
     it(`renders a link with text dependent on a given property`, () => {
       const navLink = screen.getByRole("link", {
-        name: "Go to your profile page",
+        name: linkText,
       });
 
       expect(navLink).toBeVisible();
     });
 
-    it(`renders a href for that link dependent on a given  property`, () => {
+    it(`renders a href for that link dependent on a given property`, () => {
       const navLink = screen.getByRole("link", {
-        name: "Go to your profile page",
+        name: linkText,
       });
       expect(navLink).toHaveAttribute("href", "/myProfile");
     });
 
-    it(`renders a Tooltip with an Icon, both hidden for assistive technology`, () => {
+    it(`renders an Icon`, async () => {
+      const icon = screen.getByTestId("ds-icon");
+
+      expect(icon).toBeVisible();
+    });
+
+    it(`triggers the rendering of a Tooltip on mouse hover`, async () => {
       const navLink = screen.getByRole("link", {
-        name: "Go to your profile page",
+        name: linkText,
       });
-      const tooltip = within(navLink).getByTestId("tooltip");
-      const icon = within(tooltip).getByTestId("ds-icon");
+      const queryForToolTip = () => screen.queryByRole("tooltip");
```

Comment:
Nitpicky, but I would not use a helper function here. 
`queryBy` should only be used in case you expect the element is not there, but here it's used also to asses an element is visible. 

Using `getBy`will get a more useful error response in case the element is not found. That can make debugging/fixing tests for others more comfortable later!

Reply:
I tried `getBy` but if you `get` nothing with it, the test throws an error.
```
 TestingLibraryElementError: Unable to find an accessible element with the role "tooltip"
```

 I think that if you want to assess that something has not been rendered, `queryBy` is the way :o

Reply:
Regarding the helper: I thought it was a little bit more legible than to have something like this:
```ts
const beforeHover = screen.queryByRole("tooltip")
//assert
const afterHover = screen.queryByRole("tooltip")
// assert
```

But I have no strong feeling about it ^^

Reply:
I can imagine, I had something similar recently, but in the end decided to go for `get` and `query` instead of having one helper, because it may improve DX with the better error-response a bit. Also, at least I assume an element is not supposed to be there as soon as I read `queryBy`, so it threw me off a few seconds why the element was visible :D 

So, for sake of consistency, I would do it. 

(... also, the linter I plan to introduce will otherwise flag that anyway ^^)

Reply:
>  I assume an element is not supposed to be there as soon as I read queryBy

That makes sense to me, will use `getby` when the element should be there ^^

Reply:
[refactor(coral) Test: remove queryForToolTip helper, use getByRole an…](https://github.com/aiven/klaw/pull/329/commits/abb60b46616ffe90d15879d5c9b080f683a8a9aa)